components/service/glean/src/test/java/mozilla/components/service/glean/BooleanMetricTypeTest.kt

```diff
@@ -29,8 +29,7 @@ class BooleanMetricTypeTest {
 
     @Before
     fun setUp() {
-        Glean.initialized = true
-        BooleansStorageEngine.applicationContext = ApplicationProvider.getApplicationContext()
+        resetGlean()
         // Clear the stored "user" preferences between tests.
         ApplicationProvider.getApplicationContext<Context>()
             .getSharedPreferences(BooleansStorageEngine.javaClass.canonicalName, Context.MODE_PRIVATE)
```

Comment:
We could do this for all storage engines inside of `resetGlean`, too...

Reply:
This is a great point!

Reply:
... and of course if we do that all tests start failing :-D

Reply:
Oh dear, sorry :)  I wonder why?  Certainly we can punt on that to investigate later.  This PR as is already moves the ball forward...

Reply:
Ok, I sorted it out. `FirstRunDetector` needed to be cleared as well :) This clears up things *A LOT*

Reply:
@mdboom this is RFAL!